### ISSUE TITLE
Solidify the app structure

### ISSUE THREAD

**Issue by Slesarew** [paritytech/parity-signer#883]
As development moves closer to proper release, we can solidify the navigation and content structure of the app and limit the amount of information stored by frontend side and transferred through FFI.

- [x] Navigation should be done on backend side
- [ ] All cards should be made as flat as possible, so that minimal control logic in UI is required
- [x] Only information that appears on screen should be passed around
- [ ] Optionally, decoding and navigatoin could happen in parallel thread from UI
- [ ] Replace JSONs with hard types on FFI

**Comment by krodak** [comment]
We no longer plan on moving this kind of logic to Rust side, no longer relevant
